cordyceps/src/sorted_list.rs

```diff
@@ -0,0 +1,459 @@
+where
+    T: Linked<Links<T>>,
+    T: core::cmp::Ord,
+{
+    /// Create a new (empty) sorted list, sorted LEAST FIRST
+    ///
+    /// * Consider using [`SortedList::new_max()`] if you want **largest** items sorted first.
+    /// * Consider using [`SortedList::new_custom()`] if you want to provide your own sorting
+    ///   implementation.
+    ///
+    /// If two items are considered of equal value, new values will be placed AFTER
+    /// old values.
+    pub const fn new_min() -> Self {
+        Self::new_custom(T::cmp)
+    }
+
+    /// Create a new sorted list, consuming the stack, sorted LEAST FIRST
+    pub fn from_stack_min(stack: Stack<T>) -> Self {
+        Self::from_stack_custom(stack, T::cmp)
+    }
+
+    /// Create a new (empty) sorted list, sorted GREATEST FIRST
+    ///
+    /// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+    /// * Consider using [`SortedList::new_custom()`] if you want to provide your own sorting
+    ///   implementation.
+    ///
+    /// If two items are considered of equal value, new values will be placed AFTER
+    /// old values.
+    pub const fn new_max() -> Self {
+        Self::new_custom(invert_sort::<T>)
+    }
+
+    /// Create a new sorted list, consuming the stack, sorted GREATEST FIRST
+    pub fn from_stack_max(stack: Stack<T>) -> Self {
+        Self::from_stack_custom(stack, invert_sort::<T>)
+    }
+}
+
+impl<T: Linked<Links<T>>> SortedList<T> {
+    /// Create a new (empty) sorted list with the given ordering function
+    ///
+    /// If your type T implements [`Ord`]:
+    ///
+    /// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+    /// * Consider using [`SortedList::new_max()`] if you want **largest** items sorted first.
+    ///
+    /// If `T` contained an `i32`, and you wanted the SMALLEST items at the
+    /// front, then you could provide a function something like:
+    ///
+    /// ```rust
+    /// let f: fn(&i32, &i32) -> core::cmp::Ordering = |lhs, rhs| {
+    ///     lhs.cmp(rhs)
+    /// };
+    /// ```
+    ///
+    /// SortedList takes a function (and not just [Ord]) so you can use it on types
+    /// that don't have a general way of ordering them, allowing you to select a
+    /// specific metric within the sorting function.
+    ///
+    /// If two items are considered of equal value, new values will be placed AFTER
+    /// old values.
+    pub const fn new_custom(f: fn(&T, &T) -> core::cmp::Ordering) -> Self {
+        Self {
+            func: f,
+            head: None,
+        }
+    }
+
+    /// Create a new sorted list, consuming the stack, using the provided ordering function
+    pub fn from_stack_custom(stack: Stack<T>, f: fn(&T, &T) -> core::cmp::Ordering) -> Self {
+        let mut slist = Self::new_custom(f);
+        slist.extend(stack);
+        slist
+    }
+
+    /// Pop the front-most item from the list, returning it by ownership (if it exists)
```

Comment:
it would be nice if this explicitly stated somewhere that "front-most" means based on the ordering, not based on insertion order... part of me wants to call this `pop_min` or `pop_least`, but the fact that we have a reversed-ordering constructor makes that confusing...perhaps if we did away with that and made the constructors always be min-ordered, and said "use `core::cmp::Reverse` as a wrapper if you want max-first"...but that feels more awkward...